keras_cv/layers/preprocessing/grayscale.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+import tensorflow.keras.layers as layers
+from tensorflow.keras import layers, backend
+
+
+class Grayscale(layers.Layer):
+    """Grayscale class for transforming RGB image to Grayscale image. The expected images 
+    should be [0-255] pixel ranges.
+    Input shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+    Output shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+    Args:
+        num_output_channels. It represent the output channel number after transformation. Valid values are 1 and 3.
+        (..., height, width, 1) for num_output_channels = 1 .
+        (..., height, width, 3) for num_output_channels = 3 .
+
+
+    Usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    to_grayscale = keras_cv.layers.preprocessing.ToGray()
+    augmented_images = to_grayscale(images)
+    ```
+    """
+
+    def __init__(self, num_output_channels=1, **kwargs):
+        super().__init__(**kwargs)
+        self.num_output_channels = num_output_channels
+
+    def _check_input_params(self, num_output_channels):
+        if num_output_channels not in [1, 3]:
+            raise ValueError(
+                f"Valid arugment for num_output_channels param are 1 or 3. \
```

Comment:
Received invalid argument `output_channels`.  `output_channels` must be in `1` or `3`.  Got ...

Reply:
Done.